src/BrowserProtocol.js

```diff
@@ -0,0 +1,94 @@
+import off from 'dom-helpers/events/off';
+import on from 'dom-helpers/events/on';
+import invariant from 'invariant';
+
+import createPath from './utils/createPath';
+
+export default class BrowserProtocol {
+  constructor({ basename = '' } = {}) {
+    this.basename = basename.slice(-1) === '/' ?
+      basename.slice(0, -1) : basename;
+
+    this.keyPrefix = Math.random().toString(36).slice(2, 8);
+    this.keyIndex = 0;
+
+    this.index = null;
+  }
+
+  init() {
+    const { pathname, search, hash } = window.location;
+
+    let relativePathname;
+    if (pathname.indexOf(this.basename) === 0) {
+      relativePathname = pathname.slice(this.basename.length);
+    } else {
+      relativePathname = null;
+    }
+
+    const { key, index = 0, state } = window.history.state || {};
+    const delta = this.index != null ? index - this.index : 0;
+    this.index = index;
+
+    return {
+      action: 'POP',
+      pathname: relativePathname,
+      search,
+      hash,
+      key,
+      index,
+      delta,
+      state,
+    };
+  }
+
+  subscribe(listener) {
+    const onPopState = () => {
+      listener(this.init());
+    };
+
+    on(window, 'popstate', onPopState);
```

Comment:
does Chrome not fire this on page load anymore? or does that does not matter here I guess, since redux will ask for it anyway...I guess you could get an extra stack perhaps?